site/src/pages/WorkspaceSettingsPage/WorkspaceParametersPage/WorkspaceParametersPageViewExperimental.tsx

```diff
@@ -100,16 +114,26 @@ export const WorkspaceParametersPageViewExperimental: FC<
 	) => {
 		const formInputs: Record<string, string> = {};
 		formInputs[parameter.name] = value;
-		const parameters = form.values.rich_parameter_values ?? [];
+		const values = form.values.rich_parameter_values ?? [];
 
-		for (const [fieldName, isTouched] of Object.entries(form.touched)) {
-			if (isTouched && fieldName !== parameter.name) {
-				const param = parameters.find((p) => p.name === fieldName);
-				if (param?.value) {
-					formInputs[fieldName] = param.value;
-				}
+		const touchedNames = new Set<string>();
+		values.forEach((param, idx) => {
+			const valuePath = `rich_parameter_values.${idx}.value`;
+			if (getIn(form.touched, valuePath)) {
+				touchedNames.add(param.name);
 			}
-		}
+		});
+
+		values.forEach((param) => {
+			if (
+				param?.name &&
+				param.name !== parameter.name &&
```

Comment:
`param` and `parameter` have basically identical meaning. this doesn't give me any idea what the code is supposed to be doing here.